### PR TITLE
[TS] Part 16: More Aggressive TargetedSweeper Tests

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**: Add more aggressive tests for the targeted sweeper. Specifically, this includes tests for the following scenarios:

1. sweep timestamp does not lie precisely on a coarse bucket boundary
2. some values other than the latest must be preserved
3. changes in the immutable / unreadable timestamp (including going backwards) - which also influences point 2.

**Implementation Description (bullets)**:
- Add functionality to run sweep at a specific timestamp in the tests
- Add tests that cover each of the three scenarios above
- Refactor the TargetedSweeperTest infrastructure slightly (add `as()` to make it easier to debug when a value isn't there, and also `assertTestValueEnqueuedAtGivenTimestampStillPresent` because I got annoyed at writing `assertReadAtTimestampReturnsValue(blah, x + 1, x)` everywhere)

**Concerns (what feedback would you like?)**:
- The tests for situation 3 don't feel comprehensive enough even after this change (though it does add some)

**Where should we start reviewing?**: `TargetedSweeperTests`

**Priority (whenever / two weeks / yesterday)**: Ideally this week (test-only, but it's small).

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/3195)

### CODE REVIEW

**Comment by jeremyk-91** [comment]
Note: Failing tests because of the canSweepAtMinimumTimestamp, but shouldn't fail after #3191

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java [hunk 1]
```diff
@@ -317,6 +335,148 @@ public void sweepableCellsGetsScrubbedWheneverLastSweptInNewPartition() {
...
+        enqueueWrite(TABLE_CONS, sweepTimestamp + 5);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertReadAtTimestampReturnsSentinel(TABLE_CONS, sweepTimestamp - 5);
+        assertReadAtTimestampReturnsTombstoneAtTimestamp(TABLE_CONS, sweepTimestamp - 5 + 1, sweepTimestamp - 5);
+        assertTestValueEnqueuedAtGivenTimestampStillPresent(TABLE_CONS, sweepTimestamp + 5);
+    }
+
+    @Test
+    public void sweepsOnlySweepableSegmentOfFinePartitions() {
```
Comment (gmaretic) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java]: 🏅

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java [hunk 2]
```diff
@@ -317,6 +335,148 @@ public void sweepableCellsGetsScrubbedWheneverLastSweptInNewPartition() {
...
+        assertTestValueEnqueuedAtGivenTimestampStillPresent(TABLE_CONS, newPartitionFirstTs);
+        assertTestValueEnqueuedAtGivenTimestampStillPresent(TABLE_CONS, newPartitionSecondTs);
+
+        runConservativeSweepAtTimestamp(newPartitionSecondTs + 1);
+        assertReadAtTimestampReturnsSentinel(TABLE_CONS, newPartitionSecondTs);
+        assertTestValueEnqueuedAtGivenTimestampStillPresent(TABLE_CONS, newPartitionSecondTs);
+    }
+
+    @Test
+    public void doesNotGoBackwardsEvenIfSweepTimestampRegressesWithinBucket() {
```
Comment (gmaretic) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java]: Added a check that progress is as expected

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java [hunk 3]
```diff
@@ -317,6 +335,148 @@ public void sweepableCellsGetsScrubbedWheneverLastSweptInNewPartition() {
...
+        assertReadAtTimestampReturnsSentinel(TABLE_CONS, LOW_TS2);
+        assertTestValueEnqueuedAtGivenTimestampStillPresent(TABLE_CONS, LOW_TS2);
+        verify(spiedKvs, times(1)).deleteAllTimestamps(any(TableReference.class), anyMap());
+
+        runConservativeSweepAtTimestamp(LOW_TS2 - 5);
+        verify(spiedKvs, times(1)).deleteAllTimestamps(any(TableReference.class), anyMap());
+    }
+
+    @Test
+    public void canSweepAtMinimumTime() {
```
Comment (gmaretic) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java]: Added progress check

**Pull request merged**